### DISCUSSION TITLE
Advanced building placement tags (adjecency)

### DISCUSSION THREAD

**Discussion by ayylmaoRotE** [Phobos-developers/Phobos#954] [Ideas and feedback]
### Description:
Implement tags that separate adjacency requirements for placement, e.g.:
[GAPOWR]
Adjacent.MaxDistance=4
Adjacent=6

[GAPILE]
Adjacent=10

GAPOWR can be placed 6 cells away from any building, but only 4 cells from another GAPOWR.
Same as GAPILE, can be placed up to 10 cells away but only 4 cells from GAPOWR.

### Possible Implementation:

```ini
[BuildingType]
Adjacent.MaxDistance= ;should only be applied to buildings with BaseNormal=yes and 
                      ;creates an Adjacent= restriction for all other buildings being placed around.
```

**Comment by ayylmaoRotE** [comment]
Updating some code for clarification:
what about BaseNormal.Bonus=x ;number of cells additional to the Adjacent= of the building being placed
and BaseNormal.BonusMax=x ;Adjacent cannot exceed this amount

in this scenario, it could work like:
All buildings = Adjacent=2
```
Con Yard =
BaseNormal.Bonus=8
BaseNormal.BonusMax=10
```
```
Expansion Post = 
BaseNormal.Bonus=2
BaseNormal.BonusMax=4
```
```
Pill Box = 
BaseNormal.BonusIgnore=yes
Adjacent=7
```
buildings placed around CY will have 8 cells added to their current Adjacent=2, but limited to 10
buildings placed around EP will have 2 cells added to their current Adjacent=2, but limited to 4
the pill box has Adjacent=7, it can be placed a maximum of 10 cells away from a CY but can be placed 7 cells away from a EP because it ignores the EP's 4 cell maximum via BaseNormal.BonusIgnore

**Comment by ghost** [comment]
I need this too for naval shipyards. They have big Adjacent= to reach water quicker but if they also have BaseNormal=yes then the player can jump-build over the whole ocean so a TypeAdjacent=<n> would be very useful to limit shipyard-to-shipyard expansion. I suggested the name thinking of TypeImmune= used by some units

```
[BuildingType]
TypeAdjacent=<n> ; allow placing this building n cells away from a building of the same type
                                ; this must override Adjacent=
```
